core/src/test/scala/ai/lum/odinson/foundations/TestQueryCompiler.scala

```diff
@@ -27,54 +46,182 @@ class TestQueryCompiler extends BaseSpec {
+  // TODO: refactor (DRY)
+  object QCHelper {
+    // defaults
+    // TODO: get these values from config
+    def defaultTokenField = "norm"
+    def sentenceLengthField = "numWords"
+    def dependenciesField = "dependencies"
+    // terms
+    def termFoo = new Term(defaultTokenField, "foo")
+    def termBar = new Term(defaultTokenField, "bar")
+    def termFoobar = new Term(defaultTokenField, "foobar")
+    def term(s: String) = new Term(defaultTokenField, s)
+    def termIncomingNsubj = new Term("incoming", "nsubj")
+    def termOutgoingNsubj = new Term("outgoing", "nsubj")
+    // queries
+    def spanTermQuery(t: Term) = new SpanTermQuery(t)
+    def lookaheadQuery(q: OdinsonQuery) = new LookaheadQuery(q)
+    def allNGRams0 = new AllNGramsQuery(defaultTokenField, sentenceLengthField, 0)
+    // matchers
+    def nsubjExact = new ExactLabelMatcher("nsubj", 0)
+    // query wraper
+    def wrapQuery(q: SpanTermQuery) = new OdinQueryWrapper(q)
+    def wrapQuery(q: FieldMaskingSpanQuery) = new OdinQueryWrapper(q)
+    // wraped queries
+    def wrapedFooQuery = wrapQuery(spanTermQuery(termFoo))
+    def wrapedBarQuery = wrapQuery(spanTermQuery(termBar))
+    def wrapedFoobarQuery = wrapQuery(spanTermQuery(termFoobar))
+    // wraped mask
+    def maskQuery(q: SpanTermQuery) = new FieldMaskingSpanQuery(q, defaultTokenField)
+    def wrapedMaskedIncomingNsubj: OdinsonQuery =
+      wrapQuery(maskQuery(spanTermQuery(termIncomingNsubj)))
+    def wrapedMaskedOutgoingNsubj: OdinsonQuery =
+      wrapQuery(maskQuery(spanTermQuery(termOutgoingNsubj)))
+    // graph traversal
+    def outgoingNsubj: GraphTraversal = new Outgoing(nsubjExact)
+    // full traversal
+    def fullTraversalOutgoingNsubj = new FullTraversalQuery( List( (outgoingNsubj, wrapedMaskedIncomingNsubj) ) )
+    //def fullTraversalOutgoingNsubj = FullTraversalQuery( List(outgoingNsubj, wrapedMaskedIncomingNsubj).toList)
+    // argument query
+    def argumentObjectOutgoingNsubjQuery = new ArgumentQuery("object", None, 1, Some(1), fullTraversalOutgoingNsubj)
+    // event queries
+    def eventTestQuery: OdinsonQuery = 
+      new OdinsonEventQuery(
+        triggerBarWithOutgoingNsubj,  
+        List(argumentObjectOutgoingNsubjQuery), 
+        List(), 
+        dependenciesField, 
+        defaultTokenField
+      )
+    // trigger query
+    def triggerBarWithOutgoingNsubj = new OdinsonSpanContainingQuery(wrapedBarQuery, wrapedMaskedOutgoingNsubj)
+  }
+  "OdinsonQueryCompiler" should "mkLabelMatcher should work correctly" in {
+    // get fixture
+    val ee = getExtractorEngine
+    val qc = ee.compiler
+    //
+    //qc.mkLabelMatcher(QCHelper.outgoingNsubj)
+
+  }
+
   "OdinsonQueryCompiler" should "compile beginning and end markers correctly" in {
     // get fixture
     val ee = getExtractorEngine
     val qc = ee.compiler
     // test start
-    qc.mkQuery("<s>").toString shouldEqual ("DocStartQuery")
+    qc.mkQuery("<s>") shouldEqual (new DocStartQuery("norm")
```

Comment:
For this type of thing, where you want to check that X is an instance of Y, you can use this:
https://www.scalatest.org/user_guide/using_matchers#checkingAnObjectsClass